File: src/leetcode/minimum-number-of-moves-to-seat-everyone/minimum-number-of-moves-to-seat-everyone.js
/**
 * Leetcode #2037
 * Difficulty: Easy
 * https://leetcode.com/problems/minimum-number-of-moves-to-seat-everyone/
 */
/**
 * @param {number[]} seats
 * @param {number[]} students
 * @return {number}
 */
export const minMovesToSeat = (seats, students) => {
  seats.sort((a, b) => a - b);
  students.sort((a, b) => a - b);

  let ans = 0;

  for (let i = 0; i < seats.length; i += 1) {
    ans += Math.abs(seats[i] - students[i]);
  }

  return ans;
};

File: src/leetcode/minimum-number-of-moves-to-seat-everyone/minimum-number-of-moves-to-seat-everyone.rs
impl Solution {
    pub fn min_moves_to_seat(mut seats: Vec<i32>, mut students: Vec<i32>) -> i32 {
        let mut ans = 0;

        seats.sort();
        students.sort();

        for i in 0..seats.len() {
            ans += i32::abs(seats[i] - students[i]);
        }

        ans
    }
}

// impl Solution {
//     pub fn min_moves_to_seat(mut seats: Vec<i32>, mut students: Vec<i32>) -> i32 {
//         let mut ans = 0;
//
//         seats.sort_unstable();
//         students.sort_unstable();
//
//         seats
//             .iter()
//             .zip(students.iter())
//             .map(|(seat,student)| (seat - student).abs())
//             .sum()
//     }
// }
